pipeline-scripts/buildlib.groovy

```diff
@@ -111,13 +111,22 @@ def cleanWhitespace(cmd) {
     )
 }
 
+// Preparing venv for ART tools (doozer and elliott)
+// The following commands will run automatically every time one of our jobs
+// loads buildlib (ideally, once per pipeline)
+VENV = "${env.WORKSPACE}/art-venv"
+commonlib.shell(script: "python3 -m venv --symlinks ${VENV}")
```

Comment:
I suspect we would want:
```suggestion
commonlib.shell(script: "python3 -m venv --system-site-packages --symlinks ${VENV}")
```
... in order to get the system `certifi`; also that would allow us to install fewer things in the venv.

The downside is that we wouldn't get a newer version of deps unless we specified newer versions in requirements.txt, so we might not test locally with what we get on buildvm. Maybe that just means we need to put explicit versions in requirements? Really we'd have the same problem with venvs anyway, they won't get updates unless we're explicit...

Reply:
Also... if PyPI is down/blocked... pip3 would fail trying to reach it right? Even if there's nothing to update, is there a way to try to figure that out that doesn't fail? It would be nice to ignore PyPI outages.

Maybe it's as simple as ignoring whether these succeed or not (presumably doozer/elliott will immediately choke if missing deps).